docs/RFCs/package-toml.md

```diff
@@ -0,0 +1,125 @@
+# Package.toml format
+
+A proposal for a more formal Package.toml definition
+
+- Version: DRAFT
+- Date: 06/06/2019
+- Authors
+  - Todd Kennedy <todd@selfassembled.org>
+
+## Abstract
+
+This document is meant to describe the package manifest format for Entropic,
+a federated solution to package management. Entropic needs to have its own
+manifest file due to conflicts arising from the namespacing entropic uses.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Copyright
+
+This document is copyright by its individual contributors, and available for
+re-use under [CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)
+
+## Introduction
+
+Entropic's cannot share a manifest format with the traditional `package.json` format
+associated with the ECMAScript ecosystem for a number of reasons:
+
+1. Entropic is intended to allow for federated package management for more than just the ECMAScript ecosystem
+1. The `package.json` name specifier does not allow for the namespacing that entropic provides
+1. There are recognized deficiencies with using JSON as human-readable metadata format
+
+## Filename
+
+The filename of the manifest **MUST** be `Package.toml` and it **MUST** live in
```

Comment:
There's some chatter about this in #147

Reply:
I have serious concerns about this name, especially since [pnpm released support for .yaml and .json5](https://twitter.com/pnpmjs/status/1124733887056424963), and having them support `.toml` is not a stretch. The problems caused by case-normalizing filesystems like HFS+ (with default settings) and NTFS are indeed very serious and they, for example, can seriously mess with tooling like git.

All that said, I understand the Capitalized precedent of so many other modern package managers I can think of, and for that reason, I think having something _like_ this is a good idea.

I'd like to throw my hat in for `Entropic.toml`/`Entropic.lock` as the file pair that this spec uses, if only to distance it _further_ from being a JS-only dependency specifier, but also because a completely different name completely avoids any present or future conflicts in the name style.

Reply:
Naming it `[P|p]ackage.toml` sounds like a good call if you're looking to keep CLI api compatibility with npm / yarn / etc. I don't believe that's one of your goals, though?

The first letter of a significant filename needing to be lowercase has made my life easier 100% of the times I've encountered it.

And as someone interested in entropic's model primarily for non-js packages (in the theoretical future where that's supported) - naming is `entropic.toml` (rather than package.toml, presumably for it's just connection) is also a bonus for me.

(aside: let me know if there's some pre-condition for commenting on this PRs, and if so I'll delete this)